File: Decorato.java/src/com/parttern/decorator/MatchaMilkTea.java
package com.parttern.decorator;

public class MatchaMilkTea extends MilkTea{
    private double price = 1;

    @Override
    public String doMilkTea() {
        return "Trà sữa Matcha";
    }

    @Override
    public double cost() {
        return this.price;
    }
}

File: Decorator/MatchaMilkTea.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Decorator {
    internal class MatchaMilkTea : MilkTea {
        private readonly double price = 1;

        public override string DoMilkTea() {
            return "Trà sữa Matcha";
        }

        public override double Cost() {
            return price;
        }
    }
}
